compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -139,6 +142,22 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
 
     }
 
+    @Override
+    public Optional<Symbol> symbol(Node node) {
+        if (node == null) {
```

Comment:
when can the passed node become null? can the node become null at any point?

Reply:
Removed this check. There shouldn't be a case where a syntax node becomes null